wasabi/tables.py

```diff
@@ -78,45 +79,45 @@ def table(
         "fg_colors": fg_colors,
         "bg_colors": bg_colors,
     }
-    divider_row = row(["-" * width for width in widths], **settings)
+    divider_row = row(["-" * width for width in widths], **settings)  # type: ignore
     rows = []
     if header:
-        rows.append(row(header, **settings))
+        rows.append(row(header, **settings))  # type: ignore
         if divider:
             rows.append(divider_row)
     for i, item in enumerate(data):
-        rows.append(row(item, **settings))
+        rows.append(row(item, **settings))  # type: ignore
     if footer:
         if divider:
             rows.append(divider_row)
-        rows.append(row(footer, **settings))
+        rows.append(row(footer, **settings))  # type: ignore
     return "\n{}\n".format("\n".join(rows))
 
 
 def row(
-    data,
-    widths="auto",
-    spacing=3,
-    aligns=None,
-    env_prefix="WASABI",
-    fg_colors=None,
-    bg_colors=None,
-):
+    data: Collection,
+    widths: Union[Sequence[int], int, Literal["auto"]] = "auto",
+    spacing: int = 3,
+    aligns: Optional[Union[Sequence[Literal["r", "c", "l"]], str]] = None,
+    env_prefix: str = "WASABI",
+    fg_colors: Optional[Sequence] = None,
+    bg_colors: Optional[Sequence] = None,
+) -> str:
     """Format data as a table row.
 
-    data (iterable): The individual columns to format.
-    widths (list, int or 'auto'): Column widths, either one integer for all
+    data (Collection): The individual columns to format.
+    widths (list of int, int or 'auto'): Column widths, either one integer for all
         columns or an iterable of values. If "auto", widths will be calculated
         automatically based on the largest value.
     spacing (int): Spacing between columns, in spaces.
-    aligns (list / unicode): Column alignments in order. 'l' (left,
+    aligns (Sequence / str): Optional column alignments in order. 'l' (left,
         default), 'r' (right) or 'c' (center). If a string, value is used
         for all columns.
-    env_prefix (unicode): Prefix for environment variables, e.g.
+    env_prefix (str): Prefix for environment variables, e.g.
         WASABI_LOG_FRIENDLY.
-    fg_colors (list): Foreground colors for the columns, in order. None can be
+    fg_colors (list): Optional foreground colors for the columns, in order. None can be
         specified for individual columns to retain the default foreground color.
-    bg_colors (list): Background colors for the columns, in order. None can be
+    bg_colors (list): Optional background colors for the columns, in order. None can be
```

Comment:
When a type is `Optional`, I wrote the docstring where the first word starts with "Optional...".

Reply:
Let's use the actual types in the docstrings rather than relying on the description?

Reply:
Ok will do that! There's also a type of docstring where if the input type is `Union[TypeA, TypeB]`, then I write `param (type_a / type_b)` in the docstring. I guess for clarity, I will copy over what's in the docstring?

Reply:
I think `(str / int)` is okay, but `Optional` has a specific meaning as a type that's not as clear as part of the description.

Reply:
Oh I just saw your last comment! Oops, I've already changed them to `Union[A, B]` in the docstrings (https://github.com/explosion/wasabi/pull/22/commits/1ef7d56e08c06a2b111f46c57d5eaf0e9c42b221). I can revert them back! Let me know if I should do that instead.